examples/Kittens/Sample/ViewController.m

```diff
@@ -105,4 +112,33 @@ - (BOOL)tableView:(UITableView *)tableView shouldHighlightRowAtIndexPath:(NSInde
   return NO;
 }
 
+- (BOOL)shouldBatchFetchForTableView:(UITableView *)tableView
+{
+  return _kittenDataSource.count < kMaxLitterSize;
+}
+
+- (void)tableView:(UITableView *)tableView beginBatchFetchingWithContext:(ASBatchContext *)context
+{
+  NSLog(@"adding kitties");
+  dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+    sleep(1);
+    dispatch_async(dispatch_get_main_queue(), ^{
+      NSArray *moarKittens = [self createLitterWithSize:kLitterBatchSize];
+
+      NSMutableArray *indexPaths = [[NSMutableArray alloc] init];
+      NSInteger existingKittens = _kittenDataSource.count;
+      for (NSInteger i = 0; i < moarKittens.count; i++) {
+        [indexPaths addObject:[NSIndexPath indexPathForRow:existingKittens + i inSection:0]];
```

Comment:
There's a special-case first row taking up the index path {0, 0}. So the starting position (`existingKittens`) is technically off by 1 row.